### PR TITLE
Mpr reports

### PR DESCRIPTION
**Pull request by lwyszomi**
@nickpell 

Hey, I added ASR report and small changes for MPR report.

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 7 errors:

* custom/icds_reports/mpr_sqldata.py, line 1210 - E501 line too long (120 > 115 characters)
* custom/icds_reports/mpr_sqldata.py, line 2298 - W292 no newline at end of file
* custom/icds_reports/utils.py, line 138 - W292 no newline at end of file
* custom/icds_reports/asr_sqldata.py, line 75 - E502 the backslash is redundant between brackets
* custom/icds_reports/mpr_sqldata.py, line 1206 - E501 line too long (120 > 115 characters)
* custom/icds_reports/mpr_sqldata.py, line 509 - E501 line too long (117 > 115 characters)
* custom/icds_reports/reports/reports.py, line 65 - W292 no newline at end of file

File: custom/icds_reports/reports/reports.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        config = self.report_config
+        return [
+            ASRIdentification(config=config),
+            ASROperationalization(config=config),
+            ASRPopulation(config=config),
+            Annual(config=config),
+            DisabledChildren(config=config),
+            Infrastructure(config=config),
+            Equipment(config=config)
+        ]
```
Comment (dimagimon) [inline custom/icds_reports/reports/reports.py]: W292 no newline at end of file

File: custom/icds_reports/mpr_sqldata.py [hunk 2]
```diff
@@ -0,0 +1,2307 @@
...
+    @property
+    def subtitle(self):
+        if self.config['location_id']:
+            data = self.custom_data(selected_location=self.selected_location, domain=self.config['domain'])
+            children_completing = data.get('open_child_count', 0)
+            vaccination = data.get('open_child_1yr_immun_complete', 1)
+            immunization = "%.2f%%" % ((children_completing / (vaccination or 1)) * 100)
+            return (
+                '(I) Children Completing 12 months during the month: {0}'.format(children_completing),
+                '(II) Of this, number of children who have rec
```
Comment (nickpell) [inline custom/icds_reports/mpr_sqldata.py]: http://stackoverflow.com/questions/10678229/how-can-i-selectively-escape-percent-in-python-strings

File: custom/icds_reports/asr_sqldata.py [hunk 3]
```diff
@@ -0,0 +1,1158 @@
...
+            rows = []
+            for loc_type in ['State', 'District', 'Block']:
+                loc = chosen_location.filter(location_type__name=loc_type.lower())
+                if len(loc) == 1:
+                    rows.append([loc_type, loc[0].name, loc[0].site_code])
+                else:
+                    rows.append([loc_type, '', ''])
+            supervisor = [
+                s for s in chosen_location.filter(location_type__name='supervisor')
+                if 'test' not in loc.metadata and loc.metadata.get('test', '').lower() != 'yes'
```
Comment (nickpell) [inline custom/icds_reports/asr_sqldata.py]: should this be ```loc[0]```?

File: custom/icds_reports/mpr_sqldata.py (hunk 2 above)
Comment (nickpell) [inline custom/icds_reports/mpr_sqldata.py]: can ignore

File: custom/icds_reports/asr_sqldata.py (hunk 3 above)
Comment (lwyszomi) [inline custom/icds_reports/asr_sqldata.py]: @nickpell 

Good catch :) Thanks. we should have ```s``` in conditions.

**Comment by lwyszomi** [comment]
Updated

**Pull request merged**